pkg/controller/logstorage/secrets/secret_controller.go

```diff
@@ -321,78 +334,64 @@ func (r *SecretSubController) Reconcile(ctx context.Context, request reconcile.R
-		// tigera-operator namespace.
-		esDNSNames := dns.GetServiceDNSNames(render.ElasticsearchServiceName, render.ElasticsearchNamespace, r.clusterDomain)
-		elasticKeyPair, err := cm.GetOrCreateKeyPair(r.client, render.TigeraElasticsearchInternalCertSecret, common.OperatorNamespace(), esDNSNames)
-		if err != nil {
-			r.status.SetDegraded(operatorv1.ResourceCreateError, "Failed to create Elasticsearch secrets", err, log)
-			return nil, err
-		}
-		collection.elastic = elasticKeyPair
-
-		if !r.multiTenant {
-			if kibana {
-				// Generate a keypair for Kibana.
-				//
-				// This fetches the existing key pair from the tigera-operator namespace if it exists, or generates a new one in-memory otherwise.
-				// It will be provisioned into the cluster in the render stage later on.
-				kbDNSNames := dns.GetServiceDNSNames(render.KibanaServiceName, render.KibanaNamespace, r.clusterDomain)
-				kibanaKeyPair, err := cm.GetOrCreateKeyPair(r.client, render.TigeraKibanaCertSecret, common.OperatorNamespace(), kbDNSNames)
-				if err != nil {
-					log.Error(err, err.Error())
-					r.status.SetDegraded(operatorv1.ResourceCreateError, "Failed to create Kibana secrets", err, log)
-					return nil, err
-				}
-				collection.kibana = kibanaKeyPair
-
-				// Add the es-proxy certificate to the collection. This is needed in case the user has enabled Kibana with mTLS.
-				cert, err := cm.GetCertificate(r.client, render.ManagerInternalTLSSecretName, render.ManagerNamespace)
-				if err != nil && !errors.IsNotFound(err) {
-					return nil, err
-				} else {
-					collection.upstreamCerts = append(collection.upstreamCerts, cert)
-				}
-			}
-		}
+	// Generate a keypair for elasticsearch.
+	//
+	// This fetches the existing key pair from the tigera-operator namespace if it exists, or generates a new one in-memory otherwise.
+	// It will be provisioned into the cluster in the render stage later on.
+	// Elasticsearch is always in the tigera-elasticsearch namespace, and is shared across tenants, so should always be stored in the
+	// tigera-operator namespace.
+	esDNSNames := dns.GetServiceDNSNames(render.ElasticsearchServiceName, render.ElasticsearchNamespace, r.clusterDomain)
+	elasticKeyPair, err := cm.GetOrCreateKeyPair(r.client, render.TigeraElasticsearchInternalCertSecret, common.OperatorNamespace(), esDNSNames)
+	if err != nil {
+		r.status.SetDegraded(operatorv1.ResourceCreateError, "Failed to create Elasticsearch secrets", err, log)
+		return nil, err
 	}
+	collection.elastic = elasticKeyPair
 
-	linseedNamespaces := []string{render.ElasticsearchNamespace}
-	if r.multiTenant {
-		// For multi-tenant systems, linseed runs in multiple namespaces, so we need to collect the certificates from all namespaces
-		// that have a tenant.
-		var err error
-		linseedNamespaces, err = utils.TenantNamespaces(context.Background(), r.client)
-		if err != nil {
-			return nil, err
+	if !r.multiTenant {
```

Comment:
I think I can remove this check - internal ES (which this branch is meant for) doesn't support multi-tenant clusters.